app/src/main/java/com/example/myapp/data/MyViewModel.kt

```diff
@@ -1,38 +1,49 @@
 package com.example.myapp.data
 
 import android.app.Application
-import android.util.Log
-import androidx.compose.material3.Text
 import androidx.compose.runtime.mutableStateOf
 import androidx.lifecycle.AndroidViewModel
 import androidx.lifecycle.viewModelScope
-import androidx.lifecycle.viewmodel.compose.viewModel
 import com.example.myapp.Retrofit.ArtistData
-import com.example.myapp.Retrofit.InitArtistList
-import com.example.myapp.Retrofit.InitLiveList
 import com.example.myapp.Retrofit.LiveData
+import com.example.myapp.Retrofit.getArtistList
 import kotlinx.coroutines.launch
-import java.util.stream.Collectors
 
 class MyViewModel(private val application: Application) : AndroidViewModel(application) {
     var dataList = mutableListOf<ArtistData>()
         private set
     var dataList2 = mutableListOf<LiveData>()
         private set
     val isLoaded = mutableStateOf(false)
-    init {
-        Log.i("viewmodel", "init")
-        InitData()
-    }
-    fun InitData(){
+
+    var artistList = mutableListOf<ArtistData>()
+    val isArtistListLoaded = mutableStateOf(false)
+    var pageNum = 0
+
+//    init {
+//        Log.i("viewmodel", "init")
+//        InitData()
+//    }
+//
+//    fun InitData() {
+//        viewModelScope.launch {
+//            initArtistList {
+//                dataList = it.toMutableList()
+//                isLoaded.value = true
+//            }
+//            InitLiveList {
+//                dataList2 = it.toMutableList()
+//            }
+//        }
+//    }
+
+    fun updateArtistList(page: Int = 0, size: Int = 20, name: String = "") {
+        isArtistListLoaded.value = false
         viewModelScope.launch {
-            InitArtistList {
-                dataList = it.toMutableList()
-                isLoaded.value = true
-            }
-            InitLiveList {
-                dataList2 = it.toMutableList()
-                isLoaded.value = true
+            getArtistList(page = page, size = size, name = name) { resList, res_num ->
+                artistList = resList.toMutableList()
+                pageNum = res_num
+                isArtistListLoaded.value = true

```

Comment:
```suggestion
            getArtistList(page = page, size = size, name = name) { resList, resNum ->
                artistList = resList.toMutableList()
                pageNum = resNum
                isArtistListLoaded.value = true

```
kotlin에서 변수명은 camel case를 사용합니다.